handlers/src/lib.rs

```diff
@@ -0,0 +1,99 @@
+#[macro_use]
+extern crate serde_derive;
+
+use actix_web::FromRequest;
+use actix_web::{dev::Handler, HttpRequest, Path, Query};
+use git::{GitHelper, GitOps};
+use std::path::PathBuf;
+
+#[derive(Deserialize)]
+pub struct PathParams {
+    repo: String,
+}
+
+#[derive(Deserialize)]
+pub struct QueryParams {
+    reference: String,
+    file: String,
+}
+
+pub struct RepoHandler {
+    pub repo_root: String,
+    git_ops: Box<GitOps>,
+}
+
+impl<S> Handler<S> for RepoHandler {
+    type Result = String;
+
+    fn handle(&self, req: &HttpRequest<S>) -> Self::Result {
+        let path_params = Path::<PathParams>::extract(req).unwrap();
+        let query_params = Query::<QueryParams>::extract(req).unwrap();
+        let repo_path: PathBuf = [&self.repo_root, &path_params.repo].iter().collect();
+        let reference = format!("refs/{}", query_params.reference);
+        //TODO https://actix.rs/docs/errors/
+        self.git_ops
+            .cat_file(&repo_path, &reference, &query_params.file)
+            .unwrap()
+    }
+}
+
+impl RepoHandler {
+    pub fn new(repo_root: String) -> RepoHandler {
+        RepoHandler {
+            repo_root,
+            git_ops: Box::new(GitHelper {}),
+        }
+    }
+}
+
+#[cfg(test)]
+mod tests {
+
+    use super::RepoHandler;
+    use actix_web::http::StatusCode;
+    use actix_web::test;
+    use actix_web::{Binary, Body};
+    use git::GitOps;
+    use std::path;
+
+    struct TestGitOps {
+        res: String,
+    }
+
+    impl GitOps for TestGitOps {
```

Comment:
Ah nice one, this is actually quite nice! I love `trait` + `impl` in Rust, it's so powerful.